src/util/ecdsa.rs

```diff
@@ -453,11 +454,34 @@ impl EcdsaSig {
     }
 }
 
+impl fmt::Display for EcdsaSig {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        hex::format_hex(&self.sig.serialize_der(), f)?;
+        hex::format_hex(&[self.hash_ty as u8], f)
+    }
+}
+
+impl FromStr for EcdsaSig {
+    type Err = EcdsaSigError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes = Vec::from_hex(s)?;
```

Comment:
In 0af1c3f320dd1b341dbf8ea722c7312df1dec593:

Sorry to nit about performance again, but could you use a `[u8; 73]` here, or better, lop off the last two characters and then directly call `Signature::from_str`? Again, won't hold up merging for this.

Reply:
But if the signature is less than 73 bytes the `<[u8; 73]>::from_hex` would fail, isn't it?

Reply:
No -- you can check the implementation of `Signature::from_str` which does exactly this. (I think you're right that `<[u8;73]>::from_hex` won't work, but there is something almost-as-simple which I found by checking the implementation of rust-secp src/ecdsa/mod.rs)

Reply:
Oh @dr-orlovsky sorry, I just checked again and rust-secp uses a private method `from_hex`, which despite being suggestively named, is not part of the `FromHex` trait and is not exported. So you can't do that.

What you **can** do is split the last two characters off of the string, use `Signature::from_str` and `u8::from_hex`, which is a bit uglier especially as Rust doesn't like you splitting strings by bytes like this (and we probably need to check that the string is all-ASCII first to avoid splitting characters).

But again, I won't hold the PR up for this if you feel this is overcomplicated.

Reply:
Splitting off will require multiple checks on the string length and, more importantly, on the fact that it is a correct UTF-8 string - or will introduce a panic. Arguably this will present more computational load than just a single allocation as now.

If we will stick to using Signature-style of decoding, it will require introducing additional `from_hex` function, duplicating logic from the same private function in `secp256k1`. Both seems overkill for such a marginal or arguable performance improve.

Reply:
Checking a short string for ASCII will definitely be less computational load than an allocation, even in the optimistic case that there is no memory fragmentation. And in terms of code, all you need to do is call `str.is_ascii()`.

Reply:
Ok, will do as a follow-up PR

Reply:
Perhaps for the future, there's an even more performant way to implement this. Write `fn from_hex_bytes(hex_bytes: &[u8]) -> Result<Self, Error>` on `Signature` which decodes hex without requiring type proof of UTF-8 - all hex chars are ASCII anyway so if there's anything wrong it'll just error. Then call the byte implementation from both places. Then it becomes safe to drop last two bytes and a the optimizer should throw away redundant bounds checks if the function checks constant len at the top.

Thus no redundant checking if it's ASCII nor UTF-8 chars checks.

Reply:
@apoelstra there is no `u8::from_hex` impl (as well as `<[u8; 1]>::from_hex`), so this can't be done the way you suggested.

@Kixunil what you propose requires changes to `Signature` + new release in `rust-secp256k1` lib